src/main/java/com/redhat/devtools/intellij/tektoncd/ui/component/StartTaskDialog.java

```diff
@@ -0,0 +1,459 @@
+                setInputValue(inputSelected.name(), inputParamValueTxt.getText());
+                updatePreview();
+            }
+        });
+
+        // listener for when value in resources input combo box changes
+        inputResourcesCB.addItemListener(itemEvent -> {
+            if (itemEvent.getStateChange() == 1) {
+                // when inputResourcesCB combo box value changes, inputResourceValuesCB combo box is filled with all resources of the same type as the currentInput
+                Input currentInput = (Input) itemEvent.getItem();
+                fillInputResourceComboBox(currentInput);
+                if (currentInput.value() == null && inputResourceValuesCB.getItemCount() > 0) {
+                    setInputValue(currentInput.name(), ((Resource) inputResourceValuesCB.getItemAt(0)).name());
+                }
+            }
+        });
+
+        // listener for when value in resources value input combo box changes
+        inputResourceValuesCB.addItemListener(itemEvent -> {
+            if (itemEvent.getStateChange() == 1) {
+                // when inputResourceValuesCB combo box value changes, the new value is saved and preview is updated
+                Input inputSelected = (Input) inputResourcesCB.getSelectedItem();
+                Resource resourceSelected = (Resource) itemEvent.getItem();
+                setInputValue(inputSelected.name(), resourceSelected.name());
+                updatePreview();
+            }
+        });
+
+        // listener for when value in outputs combo box changes
+        outputsCB.addItemListener(itemEvent -> {
+            if (itemEvent.getStateChange() == 1) {
+                // when outputsCB combo box value changes, outputsResourcesCB combo box is filled with all resources of the same type as the currentOutput
+                Output currentOutput = (Output) itemEvent.getItem();
+                fillOutResourcesComboBox(currentOutput);
+                if (currentOutput.value() == null && outputsResourcesCB.getItemCount() > 0) {
+                    outputs.get(outputsCB.getSelectedIndex()).setValue(((Resource) outputsResourcesCB.getItemAt(0)).name());
+                }
+            }
+        });
+
+        // listener for when value in output resources combo box changes
+        outputsResourcesCB.addItemListener(itemEvent -> {
+            if (itemEvent.getStateChange() == 1) {
+                // when outputsResourcesCB combo box value changes, the new value is saved and preview is updated
+                int outputSelectedIndex = outputsCB.getSelectedIndex();
+                Resource resourceSelected = (Resource) itemEvent.getItem();
+                outputs.get(outputSelectedIndex).setValue(resourceSelected.name());
+                updatePreview();
+            }
+        });
+
+        // listener for when the button startTask is clicked
+        startTaskButton.addMouseListener(new MouseAdapter() {
```

Comment:
In order to be UI aligned with IntelliJ, your wrapper does not need to deal with the buttons and only create a center pane. As you need to rename the OK button from OK to Start or Start task, you can retrieve the OkAction and change the name attribute. For an example of such a dialog see: https://github.com/redhat-developer/intellij-openshift-connector/blob/master/src/main/java/org/jboss/tools/intellij/openshift/ui/service/CreateServiceDialog.java

Reply:
I guess you get the wrong example but i should have found the way to rename the OK btn. setOKButtonText and doOKAction. Are you referring to them?